bowtie2/AlignmentDocker.cwl

```diff
@@ -1,62 +1,57 @@
 #!/usr/bin/env cwl-runner
 # This tool description was generated automatically by wdl2cwl ver. 0.2
 
-{
-    "baseCommand": [],
-    "requirements": [
-        {
-            "class": "ShellCommandRequirement"
-        },
-        {
-            "class": "InlineJavascriptRequirement"
-        },
-        {
-            "class": "DockerRequirement",
-            "dockerPull": "docker"
-        }
-    ],
-    "class": "CommandLineTool",
-    "inputs": [
-        {
-            "id": "leftFastq",
-            "type": "File"
-        },
-        {
-            "id": "rightFastq",
-            "type": "File"
-        },
-        {
-            "id": "ReferenceGenome",
-            "type": "File"
-        },
-        {
-            "id": "sampleName",
-            "type": "string"
-        },
-        {
-            "id": "index",
-            "type": "string"
-        },
-        {
-            "id": "docker",
-            "type": "string"
-        }
-    ],
-    "id": "Alignment",
-    "outputs": [
-        {
-            "id": "rawSAM",
-            "type": "File",
-            "outputBinding": {
-                "glob": "$(inputs.sampleName).sam"
-            }
-        }
-    ],
-    "arguments": [
-        {
-            "valueFrom": "bowtie2-build $(inputs.ReferenceGenome.path) $(inputs.index)                     bowtie2 -q -x $(inputs.index) -1 $(inputs.leftFastq.path) -2 $(inputs.rightFastq.path) -S $(inputs.sampleName).sam",
-            "shellQuote": false
-        }
-    ],
-    "cwlVersion": "v1.0"
-}
\ No newline at end of file
+    cwlVersion: v1.0
+    baseCommand: []
+    requirements:         
+           -class: ShellCommandRequirement
+        
+        
+           -class: InlineJavascriptRequirement
+        
+        
+           -class: DockerRequirement
+             dockerPull: docker
```

Comment:
```suggestion
           - class: DockerRequirement
             dockerPull: docker
```

Reply:
Made the changes.

Now the error is into arguments section.

https://github.com/ambarishK/bio-cwl-tools/blob/release/bowtie2/AlignmentDocker.cwl

```` 
(env) ngsap2@ubuntu:~$ cwl-runner alignmentDocker.cwl 
INFO /home/ngsap2/.local/bin/cwl-runner 3.0.20200324120055
INFO Resolved 'alignmentDocker.cwl' to 'file:///home/ngsap2/alignmentDocker.cwl'
ERROR Tool definition failed validation:
alignmentDocker.cwl:7:5:     Object `alignmentDocker.cwl` is not valid because
                             tried `CommandLineTool` but
alignmentDocker.cwl:104:5:     the `arguments` field is not valid because
                                 value is a CommentedMap, expected null or array of <string or
                                 Expression or CommandLineBinding>
````

Reply:
cwl-format and cwltool --validate is complete but cwltool run fails. Error is posted in above comment.